valhalla/baldr/datetime.h

```diff
@@ -199,10 +197,9 @@ static inline std::tm iso_to_tm(const std::string& iso) {
   ss.imbue(std::locale("C"));
   ss >> std::get_time(&t, "%Y-%m-%dT%H:%M");
 
-  // Validate fields. Set tm_year to 0 if any of the year,month,day,hour,minute are invalid.
-  if (t.tm_year > 200 || t.tm_mon < 0 || t.tm_mon > 11 || t.tm_mday < 0 || t.tm_mday > 31 ||
-      t.tm_hour < 0 || t.tm_hour > 23 || t.tm_min < 0 || t.tm_min > 59) {
-    t.tm_year = 0;
+  // If parsing failed zero 0 the struct
+  if (ss.fail()) {
```

Comment:
@gknisely you can more easily test for failure by checking the failbit on the stream